app/commands/training_data/code_tags_sample/update_tags.rb

```diff
@@ -0,0 +1,31 @@
+class TrainingData::CodeTagsSample::UpdateTags
+  include Mandate
+
+  initialize_with :sample, :tags, :user
+
+  def call
+    ActiveRecord::Base.transaction do
+      sample.lock!
+
+      raise TrainingDataCodeTagsSampleLockedByAnotherUserError unless sample.lockable_by?(user)
+
+      sample.update!(tags:, status: new_status, locked_until: nil, locked_by: nil)
+    end
+  end
+
+  private
+  def new_status
+    case sample.status
+    when :untagged
+      :human_tagged
+    when :machine_tagged
+      :community_checked
+    when :human_tagged
+      :community_checked
+    when :community_checked
+      :admin_checked
+    else
+      sample.status
+    end
+  end
```

Comment:
I was unsure if we wanted the command to calculate the next status, or whether the caller should do that 🤷

Reply:
Good question. Atm this never sets it to `machine_tagged` which is a problem. And also, it means someone could replay the same request twice more and move it from `untagged` to `:admin_checked` just by that replay.

So I think we probably want to do it as part of the caller instead.